File: app/src/main/java/fr/socket/florian/dhome/network/Network.java
package fr.socket.florian.dhome.network;

import java.util.List;

import fr.socket.florian.dhome.network.model.CheckServer;
import fr.socket.florian.dhome.network.model.Device;
import fr.socket.florian.dhome.network.model.Login;
import fr.socket.florian.dhome.network.model.Tokens;
import okhttp3.RequestBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;

public interface Network {

    @GET("api/")
    Call<CheckServer> checkServer();

    @POST("api/auth/login")
    Call<Login> login(@Body RequestBody body);

    @GET("api/device/all")
    Call<List<Device>> allDevices(@Header("x-access-token") String sessionToken);

    @GET("api/auth/refreshToken")
    Call<Tokens> refreshToken(@Header("x-access-token") String refreshToken);

}

File: app/src/main/java/fr/socket/florian/dhome/network/Network.kt
package fr.socket.florian.dhome.network

import fr.socket.florian.dhome.network.model.*
import okhttp3.RequestBody
import retrofit2.Call
import retrofit2.http.*

internal interface Network {

    @GET("api/")
    fun checkServer(): Call<CheckServer>

    @POST("api/login")
    fun login(@Body body: RequestBody): Call<Login>

    @GET("api/device")
    fun allDevices(@Header("x-access-token") sessionToken: String): Call<List<Device>>

    @PUT("api/device/{id}")
    fun setDeviceStatus(@Header("x-access-token") sessionToken: String, @Path("id") id: Int, @Body body: RequestBody): Call<ErrorResponse>

    @GET("api/refreshToken")
    fun refreshToken(@Header("x-access-token") refreshToken: String): Call<Tokens>

    @GET("api/stream")
    fun createScanner(@Header("x-access-token") refreshToken: String): Call<Stream>

    @GET("api/stream/{id}")
    fun readScanner(@Header("x-access-token") refreshToken: String, @Path("id") id: String): Call<Data>

    @DELETE("api/stream/{id}")
    fun deleteScanner(@Header("x-access-token") refreshToken: String, @Path("id") id: String): Call<ErrorResponse>
}
